narwhal/primary/src/primary.rs

```diff
@@ -684,9 +682,9 @@ impl PrimaryReceiverHandler {
 
         // Now that we've got all the required certificates, ensure we're voting on a
         // current Header.
-        narwhal_round = *rx_narwhal_round_updates.borrow();
+        let narwhal_round = *rx_narwhal_round_updates.borrow();
         ensure!(
-            narwhal_round <= header.round,
+            narwhal_round.saturating_sub(self.gc_depth) <= header.round,
```

Comment:
The actual GC round is `commit round - gc depth`, which is usually lower than `narwhal round - gc depth`. As discussed, we don't have to re-use the gc depth value. A separate constant might be better?